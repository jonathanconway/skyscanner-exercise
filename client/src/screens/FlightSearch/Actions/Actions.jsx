import React from 'react';

import BpkSmallPriceAlerts from 'bpk-component-icon/sm/price-alerts';
import BpkButton from 'bpk-component-button';

import STYLES from './Actions.scss';

const c = className => STYLES[className] || 'UNKNOWN';

const Actions = () => {
  return (
    <div className={c('Actions')}>
      <div>
        <BpkButton link>Filter</BpkButton>
        <BpkButton link>Sort</BpkButton>
      </div>
      <div>
        <BpkButton link>
          <BpkSmallPriceAlerts className={c('Actions__icon')} />Price Alerts
        </BpkButton>
      </div>
    </div>
  );
};

export default Actions;
