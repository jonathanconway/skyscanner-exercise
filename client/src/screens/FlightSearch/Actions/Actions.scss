@import '~bpk-mixins';

@import '../../../theme';

.Actions {
  display: flex;
  justify-content: space-between;
  background-color: $bpk-color-white;

  button {
    color: $brand-color;
    display: inline-flex;

    &:not(:last-child) {
      margin-right: $bpk-spacing-sm;
    }
  }

  &__icon {
    fill: $brand-color;
    align-self: center;
  }
}