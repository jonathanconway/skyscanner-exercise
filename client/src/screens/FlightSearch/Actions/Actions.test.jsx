import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import Actions from './Actions';

describe('Actions', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Actions />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Actions />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});