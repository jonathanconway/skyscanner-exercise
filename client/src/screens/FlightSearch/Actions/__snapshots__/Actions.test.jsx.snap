// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should render correctly 1`] = `
<div
  className="Actions"
>
  <div>
    <BpkButton
      blank={false}
      className={null}
      destructive={false}
      disabled={false}
      featured={false}
      href={null}
      iconOnly={false}
      large={false}
      link={true}
      onClick={null}
      outline={false}
      rel={null}
      secondary={false}
      submit={false}
    >
      Filter
    </BpkButton>
    <BpkButton
      blank={false}
      className={null}
      destructive={false}
      disabled={false}
      featured={false}
      href={null}
      iconOnly={false}
      large={false}
      link={true}
      onClick={null}
      outline={false}
      rel={null}
      secondary={false}
      submit={false}
    >
      Sort
    </BpkButton>
  </div>
  <div>
    <BpkButton
      blank={false}
      className={null}
      destructive={false}
      disabled={false}
      featured={false}
      href={null}
      iconOnly={false}
      large={false}
      link={true}
      onClick={null}
      outline={false}
      rel={null}
      secondary={false}
      submit={false}
    >
      <_default
        className="Actions__icon"
      />
      Price Alerts
    </BpkButton>
  </div>
</div>
`;
